### PR TITLE
Normalize confirmation for all cmdlets that have a Force parameter

### PR DESCRIPTION
**Pull request by markcowl**

### CODE REVIEW

**Comment by markcowl** [comment]
#475 
#718 
#827 
#1255

File: src/Storage/Commands.Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 namespace Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet
 {
-    [Cmdlet(VerbsCommon.Get, StorageNouns.BlobContent, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = ManualParameterSet),
+    [Cmdlet(VerbsCommon.Get, StorageNouns.BlobContent, SupportsShouldProcess = true, DefaultParameterSetName = ManualParameterSet),
```
Comment (blueww) [inline src/Storage/Commands.Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs]: What's the default ConfirmImpact?  We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ namespace Microsoft.WindowsAzure.Commands.Storage.Blob
     /// <summary>
     /// download blob from azure
     /// </summary>
-    [Cmdlet(VerbsCommon.Set, StorageNouns.BlobContent, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = ManualParameterSet),
+    [Cmdlet(VerbsCommon.Set, StorageNouns.BlobContent, SupportsShouldProcess = true, DefaultParameterSetName = ManualParameterSet),
```
Comment (blueww) [inline src/Storage/Commands.Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs]: What's the default ConfirmImpact?  We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ namespace Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet
     using System.Security.Permissions;
     using System.Threading.Tasks;
 
-    [Cmdlet(VerbsLifecycle.Start, StorageNouns.CopyBlob, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = ContainerNameParameterSet),
+    [Cmdlet(VerbsLifecycle.Start, StorageNouns.CopyBlob, SupportsShouldProcess = true, DefaultParameterSetName = ContainerNameParameterSet),
```
Comment (blueww) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs]: What's the default ConfirmImpact?  We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs [hunk 4]
```diff
@@ -25,7 +27,7 @@ namespace Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet
     using System.Security.Permissions;
     using System.Threading.Tasks;
 
-    [Cmdlet(VerbsLifecycle.Stop, StorageNouns.CopyBlob, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = NameParameterSet),
+    [Cmdlet(VerbsLifecycle.Stop, StorageNouns.CopyBlob, SupportsShouldProcess = true, DefaultParameterSetName = NameParameterSet),
```
Comment (blueww) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs]: same as above

File: src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs [hunk 5]
```diff
@@ -89,6 +91,12 @@ public string CopyId
         }
         private string copyId;
 
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            OutputStream.ConfirmWriter = (s1, s2, s3) => ShouldContinue(s2, s3);
+        }
+
```
Comment (blueww) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs]: what's this for?

File: src/Storage/Commands.Storage/File/Cmdlet/GetAzureStorageFileContent.cs [hunk 6]
```diff
@@ -25,7 +25,7 @@ namespace Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet
     using LocalDirectory = System.IO.Directory;
     using LocalPath = System.IO.Path;
 
-    [Cmdlet(VerbsCommon.Get, LocalConstants.FileContentCmdletName, SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = LocalConstants.ShareNameParameterSetName)]
+    [Cmdlet(VerbsCommon.Get, LocalConstants.FileContentCmdletName, SupportsShouldProcess = true, DefaultParameterSetName = LocalConstants.ShareNameParameterSetName)]
     public class GetAzureStorageFileContent : StorageFileDataManagementCmdletBase
```
Comment (blueww) [inline src/Storage/Commands.Storage/File/Cmdlet/GetAzureStorageFileContent.cs]: What's the default ConfirmImpact? We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/File/Cmdlet/SetAzureStorageFileContent.cs [hunk 7]
```diff
@@ -24,7 +24,7 @@ namespace Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet
     using System.Threading.Tasks;
     using LocalConstants = Microsoft.WindowsAzure.Commands.Storage.File.Constants;
 
-    [Cmdlet(VerbsCommon.Set, LocalConstants.FileContentCmdletName, SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.High, DefaultParameterSetName = LocalConstants.ShareNameParameterSetName)]
+    [Cmdlet(VerbsCommon.Set, LocalConstants.FileContentCmdletName, SupportsShouldProcess = true, DefaultParameterSetName = LocalConstants.ShareNameParameterSetName)]
```
Comment (blueww) [inline src/Storage/Commands.Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: What's the default ConfirmImpact? We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/File/Cmdlet/StopAzureStorageFileCopy.cs [hunk 8]
```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet
 {
-    [Cmdlet(VerbsLifecycle.Stop, Constants.FileCopyCmdletName)]
+    [Cmdlet(VerbsLifecycle.Stop, Constants.FileCopyCmdletName, SupportsShouldProcess = true)]
```
Comment (blueww) [inline src/Storage/Commands.Storage/File/Cmdlet/StopAzureStorageFileCopy.cs]: What's the default ConfirmImpact? We would like to have the ConfirmImpact= low for this cmdlet.

File: src/Storage/Commands.Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs (hunk 1 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs]: The default is ```ConfirmImpact.Medium```, which means no confirmation when default powershell settings are used.   In the default case, the user will only be prompted (using ```ShouldContinue```) if the download would overwrite a local file.

File: src/Storage/Commands.Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs (hunk 2 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs]: See comment above.

File: src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs (hunk 3 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs]: See comment above.

File: src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs (hunk 4 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs]: Same

File: src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs (hunk 5 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/Blob/Cmdlet/StopAzureStorageBlobCopy.cs]: In your code, you prompt for confirmation only when there is going to be an overwrite - this ensures that, in these cases, you will use ShouldContinue, which always prompts, unless the user specifies -Force

File: src/Storage/Commands.Storage/File/Cmdlet/GetAzureStorageFileContent.cs (hunk 6 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/File/Cmdlet/GetAzureStorageFileContent.cs]: See above.  Default is medium, meanign no prompting.

File: src/Storage/Commands.Storage/File/Cmdlet/SetAzureStorageFileContent.cs (hunk 7 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: See above.

File: src/Storage/Commands.Storage/File/Cmdlet/StopAzureStorageFileCopy.cs (hunk 8 above)
Comment (markcowl) [inline src/Storage/Commands.Storage/File/Cmdlet/StopAzureStorageFileCopy.cs]: See above.

**Comment by hovsepm** [comment]
:shipit:

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/956/

File: src/ResourceManager/DataFactories/Commands.DataFactories/Pipelines/SuspendAzureDataFactoryPipelineCommand.cs [hunk 9]
```diff
@@ -40,12 +43,6 @@ public override void ExecuteCmdlet()
             }
 
             ConfirmAction(
-                Force.IsPresent,
-                string.Format(
```
Comment (bgold09) [inline src/ResourceManager/DataFactories/Commands.DataFactories/Pipelines/SuspendAzureDataFactoryPipelineCommand.cs]: @markcowl won't this remove the confirmation message?

Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/Pipelines/SuspendAzureDataFactoryPipelineCommand.cs]: @bgold09 Correct, unless ```ConfirmPreference``` is < Medium, or the user specifies -Confirm, in which case the ShouldProcess confirmation is shown.

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/980/

**Comment by markcowl** [comment]
signing job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/522/

**Pull request merged**